python/lsst/ts/tcpip/base_client_or_server.py

```diff
@@ -332,6 +333,32 @@ async def read_into(self, struct: ctypes.Structure) -> None:
             await self._close_client()
             raise
 
+    async def read_str(self) -> str:
+        """Read data ending in `TERMINATOR`.
+
+        Before returning the data, it is decoded into an UTF-8 string.
+
+        Returns
+        -------
+        str:
+            Data decoded into a string.
+        """
+        data = await self.readuntil(TERMINATOR)
+        return data.decode(encoding="utf-8", errors="strict")
+
+    async def read_json(self) -> str:
+        """Read JSON data.
+
+        Before returning the data, it is decoded using JSON.
+
+        Returns
+        -------
+        str:
+            Data decoded into a string.
```

Comment:
Please document the exceptions that this method can raise, including if the data is not valid json.

Reply:
Done and also for `read_str`.